dissect/target/tools/query.py

```diff
@@ -373,23 +373,27 @@ def main() -> None:
                 continue
 
             rs = record_output(args.strings, args.json)
-            for entry in record_entries:
+            for func_generator in record_entries:
                 try:
-                    for record_entries in entry:
-                        rs.write(modifier_func(target, record_entries))
+                    for record in func_generator:
+                        rs.write(modifier_func(target, record))
                         count += 1
                         if args.limit is not None and count >= args.limit:
                             break_out = True
                             break
+
                 except Exception as e:
-                    # Ignore errors if multiple functions
-                    if len(funcs) > 1:
-                        target.log.error(f"Exception occurred while processing output of {func}", exc_info=e)
+                    # Ignore errors if multiple functions or multiple targets
+                    if len(record_entries) > 1 or len(args.targets) > 1:
+                        likely_func_name = getattr(func_generator, "__qualname__", "unknown (chained) iterator")
+                        target.log.error("Exception occurred while processing output of %s: %s", likely_func_name, e)
```

Comment:
Can't you use `func_def.name` here?

Reply:
Could also change the `record_entries` list to contain a tuple of the generator and `func_def` if that makes it any easier.